HaxeManual/04-class-field.tex

```diff
@@ -323,3 +323,30 @@ \subsection{Static}
 \haxe{assets/StaticField.hx}
 
 Static \tref{variable}{class-field-variable} and \tref{property}{class-field-property} fields can have arbitrary initialization \tref{expressions}{expression}.
+
+
+\subsection{Extern}
+\label{class-field-extern}
+
+\since{4.0.0}
+
+The \expr{extern} keyword causes the compiler to not generate the field in the output. This can be used in combination with the \tref{\expr{inline}}{class-field-inline} keyword to force a field to be inlined (or cause an error if this is not possible). Forcing inline may be desirable in abstracts or extern classes.
+
+\trivia{\expr{:extern} metadata}{Prior to Haxe 4, this access modifier could only be applied to a field using the \expr{:extern} \tref{metadata}{lf-metadata}.}
+
+
+\subsection{Final}
+\label{class-field-final}
+
+\since{4.0.0}
+
+The \expr{final} keyword can be used on class fields with the following effects:
+
+\begin{itemize}
+	\item \expr{final function ...} to make a function non-overridable in subclasses.
+	\item \expr{final x = ...} to declare a constant that requires immediate initialization and cannot be written to.
+	\item \expr{inline final x = ...} is the same but \tref{inlines}{class-field-inline} the value wherever it is used. Only constant values can be assigned.
```

Comment:
This should probably reference what we mean by "constant values".

Reply:
I originally had a `\tref` to the constants page. But the current constants page is not constants, it's literals. So we don't have actually have a constants page yet.

Reply:
#386